VisualRiders.PointOfSale.Project/Services/StaffMembersService.cs

```diff
@@ -0,0 +1,74 @@
+﻿using AutoMapper;
+using VisualRiders.PointOfSale.Project.DTOs;
+using VisualRiders.PointOfSale.Project.Models;
+using VisualRiders.PointOfSale.Project.Repositories;
+
+namespace VisualRiders.PointOfSale.Project.Services;
+
+public class StaffMembersService
+{
+    private readonly StaffMembersRepository _repository;
+    private readonly BusinessEntitiesService _businessEntitiesService;
+    private readonly IMapper _mapper;
+
+    public StaffMembersService(StaffMembersRepository repository, BusinessEntitiesService businessEntitiesService, IMapper mapper)
+    {
+        _businessEntitiesService = businessEntitiesService;
+        _repository = repository;
+        _mapper = mapper;
+    }
+
+    public ReadStaffMemberDto Create(CreateUpdateStaffMemberDto dto)
+    {
+        var staffMember = _mapper.Map<StaffMember>(dto);
+        if (_businessEntitiesService.GetById(staffMember.BusinessEntityId) == null)
+        {
+            throw new ArgumentException();
+        }
+        
+        _repository.Add(staffMember);
+        _repository.SaveChanges();
+
+        return _mapper.Map<ReadStaffMemberDto>(staffMember);
+    }
+
+    public List<ReadStaffMemberDto> GetAll()
+    {
+        return _repository.GetAll().Select(a => _mapper.Map<ReadStaffMemberDto>(a)).ToList();
```

Comment:
You could have just passed a method reference without the lambda syntax

Reply:
Yes